localstack-core/localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -782,7 +785,7 @@ def _replace(match):
 
     regex = r"\$\{([^\}]+)\}"
     result = re.sub(regex, _replace, result)
-    return result
+    return int(result) if result.isdigit() else result
```

Comment:
nit: can a `Number` parameter be a float? If so, this check will not work

Reply:
Thanks for pointing this out, will handle the float number type as well.